otherlibs/stdune/signal.ml

```diff
@@ -1,37 +1,134 @@
+type t =
+  | Int
+  | Term
+  | Abrt
+  | Alrm
+  | Fpe
+  | Hup
+  | Ill
+  | Kill
+  | Pipe
+  | Quit
+  | Segv
+  | Usr1
+  | Usr2
+  | Chld
+  | Cont
+  | Stop
+  | Tstp
+  | Ttin
+  | Ttou
+  | Vtalrm
+  | Prof
+  | Bus
+  | Poll
+  | Sys
+  | Trap
+  | Urg
+  | Xcpu
+  | Xfsz
+  | Unknown of int
+
+let all =
+  let open Sys in
+  [ (Abrt, sigabrt)
+  ; (Alrm, sigalrm)
+  ; (Fpe, sigfpe)
+  ; (Hup, sighup)
+  ; (Ill, sigill)
+  ; (Int, sigint)
+  ; (Kill, sigkill)
+  ; (Pipe, sigpipe)
+  ; (Quit, sigquit)
+  ; (Segv, sigsegv)
+  ; (Term, sigterm)
+  ; (Usr1, sigusr1)
+  ; (Usr2, sigusr2)
+  ; (Chld, sigchld)
+  ; (Cont, sigcont)
+  ; (Stop, sigstop)
+  ; (Tstp, sigtstp)
+  ; (Ttin, sigttin)
+  ; (Ttou, sigttou)
+  ; (Vtalrm, sigvtalrm)
+  ; (Prof, sigprof)
+  ; (Bus, sigbus)
+  ; (Poll, sigpoll)
+  ; (Sys, sigsys)
+  ; (Trap, sigtrap)
+  ; (Urg, sigurg)
+  ; (Xcpu, sigxcpu)
+  ; (Xfsz, sigxfsz)
+  ]
+
 let name =
   let table =
-    let open Sys in
-    [ (sigabrt, "ABRT")
-    ; (sigalrm, "ALRM")
-    ; (sigfpe, "FPE")
-    ; (sighup, "HUP")
-    ; (sigill, "ILL")
-    ; (sigint, "INT")
-    ; (sigkill, "KILL")
-    ; (sigpipe, "PIPE")
-    ; (sigquit, "QUIT")
-    ; (sigsegv, "SEGV")
-    ; (sigterm, "TERM")
-    ; (sigusr1, "USR1")
-    ; (sigusr2, "USR2")
-    ; (sigchld, "CHLD")
-    ; (sigcont, "CONT")
-    ; (sigstop, "STOP")
-    ; (sigtstp, "TSTP")
-    ; (sigttin, "TTIN")
-    ; (sigttou, "TTOU")
-    ; (sigvtalrm, "VTALRM")
-    ; (sigprof, "PROF")
-    ; (sigbus, "BUS")
-    ; (sigpoll, "POLL")
-    ; (sigsys, "SYS")
-    ; (sigtrap, "TRAP")
-    ; (sigurg, "URG")
-    ; (sigxcpu, "XCPU")
-    ; (sigxfsz, "XFSZ")
+    [ (Abrt, "ABRT")
```

Comment:
Going via a table is weird. Why not use pattern matching?

Reply:
I don't remember to be honest. We usually have a table to be able to rerverse the map, but not here. I'll make it a pattern match.